server/src/main/java/org/elasticsearch/search/retriever/KnnRetrieverBuilder.java

```diff
@@ -52,6 +54,7 @@ public final class KnnRetrieverBuilder extends RetrieverBuilder {
     public static final ParseField QUERY_VECTOR_FIELD = new ParseField("query_vector");
     public static final ParseField QUERY_VECTOR_BUILDER_FIELD = new ParseField("query_vector_builder");
     public static final ParseField VECTOR_SIMILARITY = new ParseField("similarity");
+    public static final ParseField RESCORE_FIELD = new ParseField("rescore");
```

Comment:
My concern here is that now we have a name clash if the kNN retriever ever supports general `rescore` directly.

What do we think of naming this `rescore_vector` or `vector_rescore` (here and in all knn things)?

Reply:
> My concern here is that now we have a name clash if the kNN retriever ever supports general rescore directly.

What do you mean by general rescore? 🤔   In case the knn retriever needs rescoring, shouldn't that be done as part of a parent rescorer? 

(I'm ok with changing names, just wanted to keep this as straightforward as possible for users and do less typing 😄  )

Reply:
>  In case the knn retriever needs rescoring, shouldn't that be done as part of a parent rescorer?

That would be a global instead of a per shard. But I guess its the same thing. 

However, let's still change the names. I don't want it causing a mental clash of "general rescore" that is available elsewhere.

`rescore_vector` seems like the best option to align with `query_vector`?

Reply:
OK! I changed names to `rescore_vector` in ef8ac8cdf8a092d1ff6167458aac5daecd50e5be